diff -urt gcc-14.2.0-org/gcc/cp/module.cc gcc-14.2.0/gcc/cp/module.cc
--- gcc-14.2.0-org/gcc/cp/module.cc	2024-08-01 08:17:14 +0000
+++ gcc-14.2.0/gcc/cp/module.cc	2024-08-18 20:00:18 +0000
@@ -20659,8 +20659,7 @@
             for (unsigned again = 2; ; again--)
               {
                 fd = open (tmp_name,
-                           O_RDWR | O_CREAT | O_TRUNC | O_CLOEXEC | O_BINARY,
-                           S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH|S_IWOTH);
+                           O_RDWR | O_CREAT | O_TRUNC | O_CLOEXEC | O_BINARY);
                 e = errno;
                 if (fd >= 0 || !again || e != ENOENT)
                   break;
diff -urt gcc-14.2.0-org/gcc/pretty-print.cc gcc-14.2.0/gcc/pretty-print.cc
--- gcc-14.2.0-org/gcc/pretty-print.cc	2024-08-01 08:17:14 +0000
+++ gcc-14.2.0/gcc/pretty-print.cc	2024-08-18 20:00:35 +0000
@@ -44,6 +44,14 @@
 #include <io.h>
 #define WIN32_LEAN_AND_MEAN 1
 #include <windows.h>
+ 
+#ifndef COMMON_LVB_REVERSE_VIDEO
+#define COMMON_LVB_REVERSE_VIDEO 0x4000
+#endif
+
+#ifndef COMMON_LVB_UNDERSCORE
+#define COMMON_LVB_UNDERSCORE 0x8000
+#endif
 
 /* Write all bytes in [s,s+n) into the specified stream.
    Errors are ignored.  */
diff -urt gcc-14.2.0-org/libstdc++-v3/libsupc++/del_opa.cc gcc-14.2.0/libstdc++-v3/libsupc++/del_opa.cc
--- gcc-14.2.0-org/libstdc++-v3/libsupc++/del_opa.cc	2024-08-01 08:17:18 +0000
+++ gcc-14.2.0/libstdc++-v3/libsupc++/del_opa.cc	2024-08-18 20:01:45 +0000
@@ -24,6 +24,7 @@
 // <http://www.gnu.org/licenses/>.
 
 #include <bits/c++config.h>
+#include <malloc.h>
 
 #if !_GLIBCXX_HOSTED
 // A freestanding C runtime may not provide "free" -- but there is no
@@ -50,7 +51,7 @@
     || _GLIBCXX_HAVE_MEMALIGN
   std::free (ptr);
 #elif _GLIBCXX_HAVE__ALIGNED_MALLOC
-  _aligned_free (ptr);
+  __mingw_aligned_free (ptr);
 #else
   if (ptr)
     std::free (((void **) ptr)[-1]); // See aligned_alloc in new_opa.cc
diff -urt gcc-14.2.0-org/libstdc++-v3/libsupc++/new_opa.cc gcc-14.2.0/libstdc++-v3/libsupc++/new_opa.cc
--- gcc-14.2.0-org/libstdc++-v3/libsupc++/new_opa.cc	2024-08-01 08:17:18 +0000
+++ gcc-14.2.0/libstdc++-v3/libsupc++/new_opa.cc	2024-08-18 20:01:45 +0000
@@ -24,6 +24,7 @@
 // <http://www.gnu.org/licenses/>.
 
 #include <bits/c++config.h>
+#include <malloc.h>
 #include <stdlib.h>
 #include <stdint.h>
 #include <bit>
@@ -85,7 +86,7 @@
 #elif _GLIBCXX_HAVE__ALIGNED_MALLOC
 static inline void*
 aligned_alloc (std::size_t al, std::size_t sz)
-{ return _aligned_malloc(sz, al); }
+{ return __mingw_aligned_malloc(sz, al); }
 #elif _GLIBCXX_HAVE_MEMALIGN
 static inline void*
 aligned_alloc (std::size_t al, std::size_t sz)
